snapshot/chain_correction_export.go

```diff
@@ -0,0 +1,160 @@
+package snapshot
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/thetatoken/theta/blockchain"
+	"github.com/thetatoken/theta/common"
+	"github.com/thetatoken/theta/core"
+	"github.com/thetatoken/theta/crypto"
+	"github.com/thetatoken/theta/ledger/types"
+)
+
+type BHStack []common.Hash
+
+func (stack BHStack) push(hash common.Hash) BHStack {
+	return append(stack, hash)
+}
+
+func (stack BHStack) pop() (BHStack, common.Hash) {
+	l := len(stack)
+	if l == 0 {
+		return stack, common.Hash{}
+	}
+	return stack[:l-1], stack[l-1]
+}
+
+func (stack BHStack) isEmpty() bool {
+	return len(stack) == 0
+}
+
+func ExcludeTxs(txs []common.Bytes, exclusionTxs []string, chain *blockchain.Chain) (results []common.Bytes) {
+	for _, tx := range txs {
+		t, err := types.TxFromBytes(tx)
+		if err != nil {
+			continue
+		}
+
+		// exclude stake updating txs as well
+		if _, ok := t.(*types.DepositStakeTx); ok {
+			continue
+		}
+		if _, ok := t.(*types.WithdrawStakeTx); ok {
+			continue
+		}
+
+		hash := crypto.Keccak256Hash(tx).Hex()
+		found := false
+		for _, exclusion := range exclusionTxs {
```

Comment:
Can change exclusionTxs to a map? psudo code

if !exclusionTxMap.find(hash) {
   results = append(results, tx)
}

Reply:
yah, it makes sense, esp. when there's a lot of exclusionTxs